src/axom/inlet/Table.hpp

```diff
@@ -0,0 +1,229 @@
+   *****************************************************************************
+   */
+  std::shared_ptr<Field> addDouble(const std::string& name,
+                                   const std::string& description);
+
+  /*!
+   *****************************************************************************
+   * \brief Add a Integer Field to the input deck schema.
+   *
+   * Adds a Integer Field to the input deck schema. It may or may not be required
+   * to be present in the input deck. This creates the Sidre Group class with the
+   * given name and stores the given description. If present in the input deck the
+   * value is read and stored in the datastore. 
+   *
+   * \param [in] name Name of the Field expected in the input deck
+   * \param [in] description Description of the Field
+   *****************************************************************************
+   */
+  std::shared_ptr<Field> addInt(const std::string& name,
+                                const std::string& description);
+
+  /*!
+   *****************************************************************************
+   * \brief Add a String Field to the input deck schema.
+   *
+   * Adds a String Field to the input deck schema. It may or may not be required
+   * to be present in the input deck. This creates the Sidre Group class with the
+   * given name and stores the given description. If present in the input deck the
+   * value is read and stored in the datastore. 
+   *
+   * \param [in] name Name of the Table expected in the input deck
+   * \param [in] description Description of the Table
+   *****************************************************************************
+   */
+  std::shared_ptr<Field> addString(const std::string& name,
+                                   const std::string& description);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the required status of this Table.
+   *
+   * Set whether this Table is required, or not, to be in the input deck.
+   * The default behavior is to not be required.
+   *
+   * \param [in] isRequired Boolean value of whether Table is required
+   *****************************************************************************
+   */
+  std::shared_ptr<Table> required(bool isRequired);
+
+  /*!
+   *****************************************************************************
+   * \brief Return the required status of this Table.
+   *
+   * Return that this Table is required, or not, to be in the input deck.
+   * The default behavior is to not be required.
+   *****************************************************************************
+   */
+  bool required();
+private:
+  axom::sidre::Group* baseFieldAdd(const std::string& name,
+                                   const std::string& description);
+
+  std::string m_name;
+  std::shared_ptr<Reader> m_reader;
+  // Inlet's Root Sidre Group
+  axom::sidre::Group* m_sidreRootGroup;
```

Comment:
Do you need to store the root group?  You can always get it from the current group using something like
```cpp
m_sidreGroup->getDataStore()->getRootGroup();
```
(I didn't look it up, so it might not be 100% syntactically accurate).